tools/blocktime/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/tendermint/tendermint/rpc/client/http"
+)
+
+func main() {
+	if err := Run(); err != nil {
+		fmt.Printf("ERROR: %s", err.Error())
+	}
+}
+
+func Run() error {
+	if len(os.Args) < 2 {
+		fmt.Printf("Usage: %s <node_rpc> [query_range]\n", os.Args[0])
+		return nil
+	}
+
+	url := os.Args[1]
+	c, err := http.New(url, "/websocket")
+	if err != nil {
+		return err
+	}
+	resp, err := c.Status(context.Background())
+	if err != nil {
+		return err
+	}
+	lastHeight := resp.SyncInfo.LatestBlockHeight
+	chainID := resp.NodeInfo.Network
+	queryRange := 100
+	if len(os.Args) == 3 {
+		queryRange, err = strconv.Atoi(os.Args[2])
+		if err != nil {
+			return err
+		}
+	}
+	blockTimes := make([]time.Time, 0, queryRange)
+	for i := lastHeight - int64(queryRange); i < lastHeight; i++ {
```

Comment:
[extremely optional] extract a variable for readability

```suggestion
	firstHeight := lastHeight - int64(queryRange)
	for i := firstHeight; i < lastHeight; i++ {
```

so that it can be used here and on the lines below:

```diff
-		lastHeight-int64(queryRange),
+		firstHeight
		lastHeight,
		avgTime/1000,